core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -669,6 +728,27 @@ private[spark] class ExecutorAllocationManager(
     private val stageAttemptToExecutorPlacementHints =
       new mutable.HashMap[StageAttempt, (Int, Map[String, Int], Int)]
 
+    // to track total no. of tasks in each stage of a micro-batch (streaming use case)
+    // this will help in requesting resources by counting pending tasks in job,
+    // rather than counting pending tasks in a stage.
+    private val jobStagesToNumTasks = new mutable.HashMap[Int, Int]
```

Comment:
name is a bit vague to me... what are jobStages.  Are they stageattempts or just stageId?

Reply:
They are just StageID. I will rename it